### PR TITLE
Removed interface compatibility.

### PR DESCRIPTION
**Pull request by alarshi**
In material_model/latent_heat_melt

### CODE REVIEW

**Comment by tjhei** [comment]
Arushi, here the reaction terms are missing (need to be filled with 0).

File: include/aspect/material_model/latent_heat_melt.h [hunk 1]
```diff
@@ -230,6 +198,14 @@ namespace aspect
                                   const double pressure,
                                   const std::vector<double> &compositional_fields,
                                   const Point<dim> &position) const;
+
+        virtual
+        double
+        entropy_derivative ( const double temperature,
```
Comment (tjhei) [inline include/aspect/material_model/latent_heat_melt.h]: please remove this function

File: source/material_model/latent_heat_melt.cc [hunk 2]
```diff
@@ -649,6 +615,7 @@ namespace aspect
       this->model_dependence.compressibility = NonlinearDependence::none;
       this->model_dependence.specific_heat = NonlinearDependence::none;
       this->model_dependence.thermal_conductivity = NonlinearDependence::none;
+      // this->model_dependence.entropy_derivative_temperature = NonlinearDependence::temperature;
```
Comment (tjhei) [inline source/material_model/latent_heat_melt.cc]: I think this should be ``::none``.

Comment (alarshi) [inline source/material_model/latent_heat_melt.cc]: From what I understood in the interface class, we don't have model dependence for entropy derivative. I removed the comment.

File: include/aspect/material_model/latent_heat_melt.h (hunk 1 above)
Comment (alarshi) [inline include/aspect/material_model/latent_heat_melt.h]: This function was very long, I am using it as a private member just for this class.

**Comment by tjhei** [comment]
Okay, makes sense!

File: include/aspect/material_model/latent_heat_melt.h (hunk 1 above)
Comment (bangerth) [inline include/aspect/material_model/latent_heat_melt.h]: I think that makes sense. But you can then remove the `virtual` marker.

**Comment by alarshi** [comment]
Rebase done.

**Pull request merged**